docs/developer-env.md

```diff
@@ -0,0 +1,473 @@
+(developer-env)=
+
+# Environment setup
+
+Setting up a Nextflow development environment is a prerequisite for creating, testing, and optimizing data analysis pipelines. The steps below outline recommended tools for setting up an optimal Nextflow development environment.
+
+:::{note}
+If you are using a Windows computer, you first need to install and configure the Windows Subsystem for Linux (WSL). See {ref}`wsl` for installation instructions.
```

Comment:
Why not put this at the top of the document then?

Reply:
Windows is used by a minor fraction of users, and if you're in bioinformatics, you likely already have this installed/setup. I wanted to keep it broad before getting specific for Windows users only.

Reply:
I think I like Chris's approach here too.